lib/resty/session/file.lua

```diff
@@ -119,54 +117,36 @@ local function update_sid_exp(storage, aud_sub_key, sid, exp, current_time)
 end
 
 
-local function read_metadata(storage, audience, subject, current_time)
-  local aud_sub_key = get_meta_key(storage, audience, subject)
-  local path = get_path_meta(storage, aud_sub_key)
-  local _, res = file_get(storage, path)
+local function read_metadata(storage, name, audience, subject, current_time)
+  local aud_sub_key = meta_get_key(audience, subject)
+  local path        = get_path(storage, name, aud_sub_key)
+  local _, res      = file_get(storage, path)
   if not res then
     return nil, "not found"
   end
 
-  local pattern = ".-:.-;"
-  local sessions = {}
-
-  for s in gmatch(res, pattern) do
-    local i = find(s, ":", nil, true)
-    local sid = sub(s, 1,  i - 1)
-    local exp = sub(s, i + 1, #s - 1)
-    exp = tonumber(exp)
-    if exp > current_time then
-      sessions[sid] = exp
-    else
-      sessions[sid] = nil
-    end
-  end
-
-  return sessions
+  return meta_filter_latest_valid(res, current_time)
 end
 
 
-local function cleanup_check(storage, current_time)
+local function cleanup_check(storage, name, current_time)
   if random() > TTL_CLEAN_PROBABILITY then
     return false
   end
 
   local path = storage.path
-  local suffix = storage.suffix
   local deleted = 0
 
   log(DEBUG, "[session] expired keys cleanup initiated")
 
   for file in dir(path) do
-    if file ~= "." and file ~= ".." then
-      if #file > #suffix and file:sub(#file - #suffix + 1, #file) == suffix then
-        local attr = attributes(path .. file)
-        local exp = attr and attr.modification
-        if exp < current_time then
-          file = path .. file
-          file_delete(storage, file)
-          deleted = deleted + 1
-        end
+    if validate_file_name(storage, name, file) then
+      local attr = attributes(path .. file)
```

Comment:
perhaps check that `attr.mode == "file"`, but then that should be in `validate_file_name`, perhaps then change `validate_file_name` to `get_file_expiry` (and return nil if validation or attr.mode checks fail).

Reply:
I kept them separate to not affect the unit tests, I added validate_file_attr